internal/hcsoci/resources_lcow.go

```diff
@@ -124,5 +134,43 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, res
 		}
 	}
 
+	if coi.HostingSystem != nil && coi.Spec.Windows != nil {
```

Comment:
when will coi.HostingSystem ever `== nil` for lcow?

Reply:
This check is also done earlier in this function inside the loop over mounts. I added it as a precaution since its included there. Do you think it's safe to remove?

Reply:
Yea for LCOW we dont need these checks. It would be better to crash here since it makes no sense